File: Algorithm/Python/1000+/3561. Resulting String After Adjacent Removals.py
# try stack when meeting adjacent comparing problem
class Solution:
    def resultingString(self, s: str) -> str:
        stack = []
        for i in range(len(s)):
            if len(stack) > 0 and abs(ord(s[stack[-1]]) - ord(s[i])) in (1, 25):
                stack.pop()
            else:
                stack.append(i)
        return ''.join(s[i] for i in stack)
            

File: Algorithm/C++/1000+/3561. Resulting String After Adjacent Removals.cc
#include <string>
#include <stack>

class Solution {
public:
    std::string resultingString(std::string s) {
        std::stack<int> indices;
        for(int i=0; i < s.size(); i++) {
            if (!indices.empty()) {
                int diff = abs(s[i] - s[indices.top()]);
                if (diff == 1 || diff == 25) {
                    indices.pop();
                } else {
                    indices.push(i);
                }
            } else {
                indices.push(i);              
            }
        }
        std::string rs;
        rs.reserve(indices.size());
        while (!indices.empty()) {
            rs += s[indices.top()];
            indices.pop();
        }
        return std::string(rs.rbegin(), rs.rend());
    }
};